### PR TITLE
Updated Select2 to version 4.0.12

### PR DESCRIPTION
**Pull request by kevin-brown**
Update vendored Select2 dependency to the latest minor release (4.0.7 -> 4.0.12).

<details>
<summary>
Change log
</summary>

### Bug fixes

* Fixes incorrect offset when using the Shadow DOM and styling the `<html>` element (#5682)

### Miscellaneous

* Replace cdnjs with jsDelivr in the documentation (#5687)
* Fix incorrect provider for the automated NPM deployment (#5686)

### Bug fixes

* Fixes jQuery migrate error when getting offset when dropdownParent not in document (#5584)

### Miscellaneous

* Enable GitHub actions for CI (#5591)
* Documentation has been moved into and is deployed from the code repository (#5638)
* Remove Travis CI integration (#5665)

### New features / improvements

* Support passing in a selector for `dropdownParent` option (#5622)

### Bug fixes

* Fix bug where dropdowns pointing upwards were incorrectly positioned (#5621)

### New features / improvements

* Mirror disabled state through aria-disabled on selection (#5579)
* Select2 now clears the internal ID when it is destroyed (#5587)
* Set the main ARIA 1.1 roles and properties for comboboxes (#5582)
* The `language` option now has a clearly defined fallback chain (#5602)

### Bug fixes

* Do not propagate click when search box is not empty (#5580)
* Fix `maximumSelectionLength` being ignored by `closeOnSelect` (#5581)
* Fix generated options not receiving result IDs (#5586)
* Remove selection title attribute if text is empty (#5589)
* Reposition dropdown whenever items are selected (#5590)
* Fix dropdown positioning when displayed above with messages (#5592)
* Fix search box expanding width of container (#5595)
* `allowClear` no longer shifts selections to a new line (#5603)

### Translations

* Fix error in German translations (#5604)

### Miscellaneous

* Updated development grunt version so it no longer shows as vulnerable (#5597)
* Remove unused variables (#5554)

### New features / improvements

* Test against and fix compatibility with jQuery 3.4.1 (#5531)
* Results respect disabled state of `<option>` (#5560)
* Add `computedstyle` option for calculating the width (#5559)

### Bug fixes

* Fix tag creation being broken in 4.0.7 (#5558)
* Fix infinite scroll when the scrollbar is not visible (#5575)
* Revert change to focusing behaviour in 4.0.6 (#5576)

### Translations

* Fix wording in French translations (#5521)

### Miscellaneous

* Update grunt-contrib-qunit to latest version (#5530)
* Removed unused `.select2-selection__placeholder` CSS definitions for multiple selects (#5508)
* Remove deprecated jQuery shorthand (#5564)
</details>

### CODE REVIEW

File: docs/releases/3.0.txt [hunk 1]
```diff
@@ -106,7 +106,7 @@ Minor features
   <django.contrib.admin.ModelAdmin.get_inlines>` method allows specifying the
   inlines based on the request or model instance.
 
-* Select2 library is upgraded from version 4.0.3 to 4.0.7.
+* Select2 library is upgraded from version 4.0.3 to 4.0.12.
```
Comment (pope1ni) [inline docs/releases/3.0.txt]: Django 3.0 is in feature freeze. Can you revert this and add an entry to `docs/releases/3.1.txt`?

**Comment by felixxm** [comment]
@kevin-brown Thanks for this patch, however I think we should postpone the `Select2` update closer to the Django 3.1 feature freeze date ~ April 2021. I don't see much value in doing this multiple times.

**Comment by pope1ni** [comment]
@felixxm Is it worth re-opening this now?

**Comment by felixxm** [comment]
@pope1ni I think yes, probably with `4.0.13`. I'm not sure if `4.1.0` will be ready before the future freeze.

**Comment by kevin-brown** [comment]
I can submit a new PR with Select2 4.0.13. I agree that 4.1.0 would be cutting is close.

When is the feature freeze date for Django 3.1?

**Comment by felixxm** [comment]
Thanks, the feature freeze is on [11th, May](https://code.djangoproject.com/wiki/Version3.1Roadmap).

**Comment by felixxm** [comment]
Superseded by #12836.

**Pull request closed without merging**